File: Java/src/main/java/com/intersystems/demo/pex/NonPollingBusinessService.java
package com.intersystems.demo.pex;

import com.intersystems.gateway.JavaGateway;
import com.intersystems.jdbc.IRISConnection;
import com.intersystems.enslib.pex.Director;

public class NonPollingBusinessService {

	public static void main(String[] args) {
		
		try {
			IRISConnection connection = JavaGateway.getConnection("jdbc:IRIS://localhost:51770/PEX/","SuperUser","password");
			com.intersystems.enslib.pex.BusinessService service = Director.CreateBusinessService(connection,"Demo.PEX.Service");
			String response = (String) service.ProcessInput(String.join(" ", args));
			System.out.println("\r\n" + response);
		} catch ( Exception ex ) {
			System.out.println(ex);
		}
	
	}

}

File: DotNet/NonPollingBusinessService.cs
using System;
using InterSystems.Data.IRISClient;

namespace InterSystems.Demo.PEX
{
    class NonPollingBusinessService
    {
        static void Main(string[] args)
        {
            try
            {
                IRISADOConnection connection = InterSystems.Data.IRISClient.Gateway.Gateway.GetConnection("Server=localhost; Port=51774; Namespace=Ensemble; Password=SYS; User ID=_SYSTEM; SharedMemory=false");
                InterSystems.EnsLib.PEX.BusinessService service = InterSystems.EnsLib.PEX.Director.CreateBusinessService(connection, "Demo.PEX.NonPollingBusinessService");
                string response = (string)service.ProcessInput(String.Join(" ", args));
                Console.WriteLine("\r\n" + response);
            }
            catch (Exception ex)
            {
                Console.Write(ex);
            }
        }

    }
}
